Rollbar/DTOs/ExtendableDtoBase.cs

```diff
@@ -0,0 +1,281 @@
+        /// otherwise, false.  
+        /// This method also returns false if <paramref name="key">key</paramref> was not found 
+        /// in the original <see cref="T:System.Collections.Generic.IDictionary`2"></see>.
+        /// </returns>
+        public bool Remove(string key)
+        {
+            return this._keyedValues.Remove(key);
+        }
+
+        /// <summary>
+        /// Removes the first occurrence of a specific object from 
+        /// the <see cref="T:System.Collections.Generic.ICollection`1"></see>.
+        /// </summary>
+        /// <param name="item">
+        /// The object to remove from the <see cref="T:System.Collections.Generic.ICollection`1"></see>.
+        /// </param>
+        /// <returns>
+        /// true if <paramref name="item">item</paramref> was successfully removed from 
+        /// the <see cref="T:System.Collections.Generic.ICollection`1"></see>; 
+        /// otherwise, false. 
+        /// This method also returns false if <paramref name="item">item</paramref> is not found 
+        /// in the original <see cref="T:System.Collections.Generic.ICollection`1"></see>.
+        /// </returns>
+        public bool Remove(KeyValuePair<string, object> item)
+        {
+            return this._keyedValues.Remove(item);
+        }
+
+        /// <summary>
+        /// Gets the value associated with the specified key.
+        /// </summary>
+        /// <param name="key">The key whose value to get.</param>
+        /// <param name="value">
+        /// When this method returns, the value associated with the specified key, 
+        /// if the key is found; otherwise, the default value 
+        /// for the type of the value parameter. This parameter is passed uninitialized.
+        /// </param>
+        /// <returns>
+        /// true if the object that implements 
+        /// <see cref="T:System.Collections.Generic.IDictionary`2"></see> contains 
+        /// an element with the specified key; otherwise, false.
+        /// </returns>
+        public bool TryGetValue(string key, out object value)
+        {
+            return this._keyedValues.TryGetValue(key, out value);
+        }
+
+        /// <summary>
+        /// Returns an enumerator that iterates through a collection.
+        /// </summary>
+        /// <returns>
+        /// An <see cref="T:System.Collections.IEnumerator"></see> object 
+        /// that can be used to iterate through the collection.
+        /// </returns>
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return this.GetEnumerator();
+        }
+
+        /// <summary>
+        /// Extends this instance with the specified extra key-value pairs.
+        /// </summary>
+        /// <param name="extraProperties">The extra key-value pairs.</param>
+        public void Extend(Dictionary<string, object> extraProperties)
+        {
+            foreach (var kvp in extraProperties)
+            {
+                this._keyedValues[kvp.Key] = kvp.Value;
```

Comment:
We go directly to setting this properties in the underlying keyedValues map rather than going through the indexer for this object I am presuming because we do not want those assertions to run when we call extend?

Reply:
yes, the way indexer setter works it either modifies value for existing key or allocates a new key value pair without causing exceptions. i think that's what we need in this use case...